build-tools/xaprepare/xaprepare/Application/Utilities.cs

```diff
@@ -99,6 +99,10 @@ public static async Task<bool> BuildRemapRef (Context context, bool haveManagedR
 			bool result = await msbuild.Run (
 				projectPath: projectPath,
 				logTag: "remap-assembly-ref",
+				arguments: new List <string> {
+				   "/t:Restore",
+				   "/t:Build",
```

Comment:
I think it would be enough to use just `/restore` here? Same for other cases below.

Reply:
Nope. exe was not generated. I HAD to add it

Reply:
I was under impression that `/restore` implied `/t:Build`, too. Odd.

Reply:
Me too. So I played with commandline and after adding `/t:Build` exe was generated.

I could have added it to MsBuildRunner, but I thought it would be better to add it just for projects that need it.

Reply:
Do you have a build log of when it didn't work?

You should find `bin/BuildDebug/msbuild-*-build-remap-assembly-ref.binlog`  -- or if it was on CI, we have it.

Reply:
I have locally removed that line in Utilities.cs

```
msbuild "/p:Configuration=Debug" /v:normal "/bl:bin/BuildDebug/msbuild-20210112T154632-build-conjure-xamarin-android-cecil.binlog" "/t:Restore" "/t:Build" "build-tools/conjure-xamarin-android-cecil/conjure-xamarin-android-cecil.csproj" 
```

Console log

```
[MSBuild] running
[MSBuild] log file: bin/BuildDebug/prepare-20210112T154632.msbuild.conjure-xamarin-android-cecil.log
Elapsed: 00:00:02.0377381
  • Conjuring Xamarin.Android.Cecil and Xamari.Android.Cecil.Mdb
  • Re-signing Xamarin.Android.Cecil.dll
sn "-R" "bin/BuildDebug/Xamarin.Android.Cecil.dll" "mono.snk" 
[sn] running
[sn] log file: bin/BuildDebug/prepare-20210112T154632.sn.sign-xamarin-android-cecil.log
  • Re-signing Xamarin.Android.Cecil.Mdb.dll
sn "-R" "bin/BuildDebug/Xamarin.Android.Cecil.Mdb.dll" "mono.snk" 
[sn] running
[sn] log file: bin/BuildDebug/prepare-20210112T154632.sn.sign-xamarin-android-cecil-mdb.log
  • Installing tests
    • Creating BCL tests archive
Archive path: /Users/Shared/Projects/d/X.tmp/migration-to-sdk-short-style-projects-02-20210111/bin/Debug/bcl-tests/bcl-tests.zip
[7zip] creating zip archive
[7zip] log file: bin/BuildDebug/prepare-20210112T154632.7zip-create-zip.bcl-tests.zip.log
Elapsed: 00:00:06.3236318
  • Installing runtimes
    • Installing Android JIT runtime armeabi-v7a
    • Installing Android JIT runtime armeabi-v7a (interpreter)
    • Installing Android JIT runtime arm64-v8a
    • Installing Android JIT runtime arm64-v8a (interpreter)
    • Installing Android JIT runtime x86
    • Installing Android JIT runtime x86 (interpreter)
    • Installing Android JIT runtime x86_64
    • Installing Android JIT runtime x86_64 (interpreter)
    • Installing host runtime Darwin
    • Installing cross compilation runtime cross-arm
    • Installing cross compilation runtime cross-arm64
    • Installing cross compilation runtime cross-x86
    • Installing cross compilation runtime cross-x86_64
    • Installing LLVM runtime llvm64
  • Installing Android BCL assemblies
  • Installing Designer Host BCL assemblies
  • Installing Designer Windows BCL assemblies
  • Installing runtime utilities
    • Remapping Cecil references for bin/Debug/lib/xamarin.android/xbuild/Xamarin/Android/illinkanalyzer.exe
stderr | Cannot open assembly 'bin/BuildDebug/remap-assembly-ref.exe': No such file or directory.
  Error: Failed to remap cecil reference for /Users/Shared/Projects/d/X.tmp/migration-to-sdk-short-style-projects-02-20210111/bin/Debug/lib/xamarin.android/xbuild/Xamarin/Android/illinkanalyzer.exe

Step Xamarin.Android.Prepare.Step_InstallMonoRuntimes failed
System.InvalidOperationException: Step Xamarin.Android.Prepare.Step_InstallMonoRuntimes failed
  at Xamarin.Android.Prepare.Scenario.Run (Xamarin.Android.Prepare.Context context, Xamarin.Android.Prepare.Log log) [0x001a8] in /Users/Shared/Projects/d/X.tmp/migration-to-sdk-short-style-projects-02-20210111/build-tools/xaprepare/xaprepare/Application/Scenario.cs:50 
  at Xamarin.Android.Prepare.Context.Execute () [0x000fc] in /Users/Shared/Projects/d/X.tmp/migration-to-sdk-short-style-projects-02-20210111/build-tools/xaprepare/xaprepare/Application/Context.cs:816 
  at Xamarin.Android.Prepare.App.Run (System.String[] args) [0x00861] in /Users/Shared/Projects/d/X.tmp/migration-to-sdk-short-style-projects-02-20210111/build-tools/xaprepare/xaprepare/Main.cs:164 

make: *** [prepare] Error 1
```

[prepare-20210112T154632.msbuild.conjure-xamarin-android-cecil.log](https://github.com/xamarin/xamarin-android/files/5802869/prepare-20210112T154632.msbuild.conjure-xamarin-android-cecil.log)


Replacing `normal` with `diagnostic` (next upcoming comment)

Reply:
binlog
[msbuild-20210112T154632-build-conjure-xamarin-android-cecil.binlog.zip](https://github.com/xamarin/xamarin-android/files/5802996/msbuild-20210112T154632-build-conjure-xamarin-android-cecil.binlog.zip)